src/SDK/SdkAutoloader.php

```diff
@@ -30,80 +36,121 @@ public static function autoload(): bool
+        } else {
+            Globals::registerInitializer(fn ($configurator) => self::environmentBasedInitializer($configurator));
+        }
+        self::registerInstrumentations($sdk);
 
         return true;
     }
 
+    private static function environmentBasedInitializer(Configurator $configurator): Configurator
+    {
+        $propagator = (new PropagatorFactory())->create();
+        if (Sdk::isDisabled()) {
+            //@see https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/configuration/sdk-environment-variables.md#general-sdk-configuration
+            return $configurator->withPropagator($propagator);
+        }
+        $emitMetrics = Configuration::getBoolean(Variables::OTEL_PHP_INTERNAL_METRICS_ENABLED);
+
+        $resource = ResourceInfoFactory::defaultResource();
+        $exporter = (new ExporterFactory())->create();
+        $meterProvider = (new MeterProviderFactory())->create($resource);
+        $spanProcessor = (new SpanProcessorFactory())->create($exporter, $emitMetrics ? $meterProvider : null);
+        $tracerProvider = (new TracerProviderBuilder())
+            ->addSpanProcessor($spanProcessor)
+            ->setResource($resource)
+            ->setSampler((new SamplerFactory())->create())
+            ->build();
+
+        $loggerProvider = (new LoggerProviderFactory())->create($emitMetrics ? $meterProvider : null, $resource);
+        $eventLoggerProvider = (new EventLoggerProviderFactory())->create($loggerProvider);
+
+        ShutdownHandler::register($tracerProvider->shutdown(...));
+        ShutdownHandler::register($meterProvider->shutdown(...));
+        ShutdownHandler::register($loggerProvider->shutdown(...));
+
+        return $configurator
+            ->withTracerProvider($tracerProvider)
+            ->withMeterProvider($meterProvider)
+            ->withLoggerProvider($loggerProvider)
+            ->withEventLoggerProvider($eventLoggerProvider)
+            ->withPropagator($propagator)
+        ;
+    }
+
+    private static function createSdkFromFileConfig(): Sdk
+    {
+        $file = Configuration::getString(Variables::OTEL_EXPERIMENTAL_CONFIG_FILE);
+        $config = SdkConfiguration::parseFile($file);
+
+        return $config
+            ->create()
+            ->setAutoShutdown(true)
+            ->build();
+    }
+
     /**
-     * Test whether a request URI is set, and if it matches the excluded urls configuration option
+     * Register any auto-instrumentation configured through SPI
      *
-     * @internal
+     * @phan-suppress PhanUndeclaredClassMethod
      */
-    public static function isIgnoredUrl(): bool
+    private static function registerInstrumentations(?Sdk $sdk): void
     {
-        $ignoreUrls = Configuration::getList(Variables::OTEL_PHP_EXCLUDED_URLS, []);
-        if ($ignoreUrls === []) {
-            return false;
+        if (!Configuration::has(Variables::OTEL_PHP_INSTRUMENTATION_CONFIG_FILE)) {
+            return;
```

Comment:
IMO we should load instrumentations with empty configuration registry if the config file is not specified.

Reply:
It _seems_ to work. `autoload.files` should not be an issue since everything continues to use globals initializers.
I've wrapped registering instrumentation in a try/catch/log, since some will not work without config. If we could work out a way to provide the default config rather than empty (ie, by processing the `InstrumentationConfiguration` without input), it might work a bit better?

Reply:
> If we could work out a way to provide the default config rather than empty (ie, by processing the `InstrumentationConfiguration` without input), it might work a bit better?

This is the bit I'm watching out for too, as I'm trying to keep existing behaviour locally without providing config files.

Reply:
One solution could be to extend the `InstrumentationConfiguration` interface so that each config class can provide its own defaults:
```php
interface InstrumentationConfiguration
{
    public static function default(): self;
}

class ExampleInstrumentation implements InstrumentationConfiguration
{
    public static function default(): self
    {
        return new self('default1', 'another-default');
    }
}

public function register(HookManager $hookManager, ConfigurationRegistry $configuration, InstrumentationContext $context): void
    {
        $config = $configuration->get(ExampleConfig::class) ?? ExampleConfig::default();
        //snip
    }
}
```

Reply:
I still haven't solved this, however I wonder if we could actually set the defaults in the config provider's `getConfig()` method? That said, I guess defaults could be handled in the instrumentation code like in the example above...